### ISSUE TITLE
调用模型后输出的标签不正常

### ISSUE THREAD

**Issue by magician-m** [YeexiaoZheng/Multimodal-Sentiment-Analysis#12]
**1.佬，又来打扰你了。上次我改了一下还是有好多问题（T~T）。不过还是一定要蟹蟹你呢！感谢大佬抽时间看我这个小菜鸡的issue！**
**2.现在我遇到的问题如下：**
<mark>模型接受数据：<mark>
``# 模型输入
model_input = {
    'texts': text_input_ids,  # 将 input_ids 改为 texts
    'texts_mask': text_attention_mask,  # 将 attention_mask 改为 texts_mask
    'imgs': image_tensor,
    # 'guids' 和 'labels' 
    # 'guids': torch.tensor([guid]),
    # 'labels': torch.tensor([tokenizer.label_vocab.label_to_id(label)]),``
得到的输出：
![image](https://github.com/YeexiaoZheng/Multimodal-Sentiment-Analysis/assets/80674900/94828797-cec9-4f81-8d4d-58b2278ce234)

**3.全部代码如下：**
import torch
from PIL import Image
from torchvision import transforms
from transformers import RobertaTokenizer
from Models.OTEModel import Model  # 根据实际使用的模型进行调整
from utils.DataProcess import Processor
from utils.DataProcess import LabelVocab
# 导入配置信息，这里假设 Config 类已经定义并提供了必要的配置
from Config import config

# 初始化配置
config = config()

# 模型路径和配置
model_path = "E:/aGraduation_Program_File/Multimodal-Sentiment-Analysis-main/output/OTE/pytorch_model.bin"
model = Model(config)

# 加载模型权重
model.load_state_dict(torch.load(model_path, map_location=torch.device('cpu')))
model.eval()

# 用户输入的数据
guid = 0
label = 'null'
text = 'so fast'
image_path = 'E:/aGraduation_Program_File/Multimodal-Sentiment-Analysis-main/web/Cavendish.jpg'

# 文本预处理
tokenizer = RobertaTokenizer.from_pretrained('C:/Users/Lenovo/.cache/huggingface/hub/models--roberta-base')  # 初始化分词器
text_tokens = tokenizer.encode(text, add_special_tokens=True)
text_input_ids = torch.tensor([text_tokens])  # 转换为张量并添加批次维度
text_attention_mask = torch.ones_like(text_input_ids, dtype=torch.long)

# 图像预处理
image = Image.open(image_path)
preprocess = transforms.Compose([
    transforms.Resize((224, 224)),  # 调整图像大小为 224x224
    transforms.ToTensor(),  # 将PIL图像转换为Tensor
    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])  # 归一化
])
image_tensor = preprocess(image).unsqueeze(0)  # 添加批次维度

# 模型输入
model_input = {
    'texts': text_input_ids,  # 将 input_ids 改为 texts
    'texts_mask': text_attention_mask,  # 将 attention_mask 改为 texts_mask
    'imgs': image_tensor,
    # 'guids' 和 'labels'
    # 'guids': torch.tensor([guid]),
    # 'labels': torch.tensor([tokenizer.label_vocab.label_to_id(label)]),
}
processor = Processor(config)
label_vocab = LabelVocab()
# 使用模型进行预测
with torch.no_grad():
    outputs = model(**model_input)

print('pred_label'：',outputs)
print('输出的格式为：',type(outputs))

---

[run_model_noweb.md](https://github.com/YeexiaoZheng/Multimodal-Sentiment-Analysis/files/14602658/run_model_noweb.md)
**4.我想是不是pred_label需要进行解码，但我进行解码后还是不行。所以是哪出错了？呜呜呜...大佬，救我！**

**Comment by YeexiaoZheng** [comment]
你这里不是已经输出了标签了嘛？tensor([0])，表示这输出的标签就是0，是只需要decode就可以输出negative这些str标签，processor.decode(outputs)就可以，或者你看一下0代表的是negative、positive、neutral，直接判也可以

**Comment by magician-m** [comment]
![image](https://github.com/YeexiaoZheng/Multimodal-Sentiment-Analysis/assets/80674900/34f59c9c-a2a2-4d40-bdfe-970836595402)
感谢大佬！已经完成了！感谢感谢！我会再多学习学习佬写的代码!

**Comment by YeexiaoZheng** [comment]
哈哈哈哈 撒花！我也不是什么佬，这个模型应该还是有点问题在的，不过众所周知神经网络效果行就行，欢迎更多交流讨论。
